### PR TITLE
Add FalkorDB support

### PR DESCRIPTION
**Pull request by gkorland**
## Summary by CodeRabbit

- **New Features**
  - Introduced support for FalkorDB as a new graph database option.
  - Implemented a new FalkorDB driver for database interactions.

- **Enhancements**
  - Updated the README with instructions for database selection and setup.
  - Provided additional run command options for graph selection, port configuration, and debug mode.

- **Documentation**
  - Added guidance for running the application within a container.

- **Bug Fixes**
  - Improved the Neo4J driver initialization and data retrieval methods.

- **Refactor**
  - Streamlined methods in driver classes for better readability and consistency.

- **Chores**
  - Included new environment variables in `.env.example`.
  - Updated `requirements.txt` with the latest FalkorDB library version.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -262,6 +264,8 @@ def index():
     match graph.lower():
```
Comment (isamu) [inline main.py]: `match` cannot be used with older python, so please change it if possible.
See #84

**Comment by isamu** [comment]
Your changes is good and it is working in my environment.
There is a conflicts, please resolve it.

File: main.py (hunk 1 above)
Comment (gkorland) [reply main.py]: changed

Comment (isamu) [reply main.py]: Thanks!!

**Comment by isamu** [comment]
Thank you!!
I merged this PR.

**Pull request merged**